File: LeetCode/0796/python/lc0796_test.py
import unittest

from lc0796 import Solution

class Test_rotateString(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.rotateString("abcde", "cdeab"), True)
        self.assertEqual(s.rotateString("abcde", "abced"), False)
        self.assertEqual(s.rotateString("aa", "a"), False) # WA #01

if __name__ == "__main__":
    unittest.main()

File: LeetCode/0796/golang/lc0796_test.go
package lc0796

import "testing"

func Test_rotateString(t *testing.T) {
	testcases := []struct {
		s        string
		goal     string
		expected bool
	}{
		{"abcde", "cdeab", true},
		{"abcde", "abced", false},
		{"aa", "a", false},
	}

	for i, tc := range testcases {
		actual := rotateString(tc.s, tc.goal)
		if actual != tc.expected {
			t.Errorf("Testcase rotateString#%02d (%v) failed: want %t, got %t",
				i+1, tc.s, tc.expected, actual)
		}
	}
}
